lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -778,45 +778,22 @@ LogicalResult applyGCTMemTapsWithWires(const AnnoPathValue &target,
     return failure();
   Optional<AnnoPathValue> wireTarget = resolvePath(
       wireTargetStr, state.circuit, state.symTbl, state.targetCaches);
-  SmallVector<InstanceOp> pathFromSrcToWire;
-  FModuleOp lcaModule;
-  // Find the lca and get the path from source to wire through that lca.
-  if (findLCAandSetPath(*srcTarget, *wireTarget, pathFromSrcToWire, lcaModule,
-                        state)
-          .failed())
-    return mlir::emitError(loc,
-                           "Failed to find a uinque path from source to wire.");
-  LLVM_DEBUG(llvm::dbgs() << "\n lca :" << lcaModule.getNameAttr();
-             for (auto i
-                  : pathFromSrcToWire) llvm::dbgs()
-             << "\n"
-             << i->getParentOfType<FModuleOp>().getNameAttr() << ">"
-             << i.getNameAttr(););
-  auto srcModule =
-      dyn_cast<FModuleOp>(srcTarget->ref.getModule().getOperation());
-  ImplicitLocOpBuilder refSendBuilder(srcModule.getLoc(), srcModule);
+  if (!wireTarget)
+    return mlir::emitError(loc, "Annotation '" + Twine(memTapClass) +
+                                    "' with path '.taps[0]' contains target '" +
+                                    wireTargetStr +
+                                    "' that cannot be resolved.")
+               .attachNote()
+           << "The full Annotation is reproduced here: " << anno << "\n";
+
```

Comment:
👍 Thanks for adding this. I had to check that `.taps[0]` made sense to include and I think it does. Nice.

Reply:
I think a helper for emitting these errors would be very useful.   Avoid deviations/typos and make it less verbose to emit these errors on failure paths.

I was doubtful about the "path" portion of some of these errors earlier to be honest (confused why we appended '.class' to one, for example) but yeah I think they're useful now.  If nothing else consistency here is good :+1: .